### PR TITLE
Refactor the repository layout

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

> [!NOTE]
> This is built on top of #376 because we need the updated version of [cisagov/pre-commit-packer]. The changes for just this pull request can be viewed at https://github.com/cisagov/skeleton-packer/compare/lineage/skeleton...improvement/refactor_repository

This pull request reformats the repository layout including breaking the Packer template into multiple files.

## 💭 Motivation and context ##

With the `0.1.0` release of [cisagov/pre-commit-packer] we can now use Packer templates that consist of more than one file. Breaking up this template will make it easier to maintain and modify. As part of that it makes sense to treat Packer projects more like Terraform root modules since both Packer and Terraform are Hashicorp projects that use the Hashicorp configuration language.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

[cisagov/pre-commit-packer]: https://github.com/cisagov/pre-commit-packer

### CODE REVIEW

**Review by dv4harr10 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -129,12 +117,12 @@ Here is an example of how to kick off a pre-release build:
...
 pip install --requirement requirements-dev.txt
-ansible-galaxy install --force --force-with-deps --role-file src/requirements.yml
-AWS_PROFILE=cool-images-ec2amicreate-skeleton-packer packer build --timestamp-ui -var release_tag=$(./bump_version.sh show) -var is_prerelease=true src/packer.pkr.hcl
+ansible-galaxy install --force --force-with-deps --role-file ansible/requirements.yml
+AWS_PROFILE=cool-images-ec2amicreate-skeleton-packer packer build --timestamp-ui -var release_tag=$(./bump_version.sh show) -var is_prerelease=true packer.pkr.hcl
 ```
 
 If you are satisfied with your pre-release image, you can easily create a release
-that deploys to all regions by adding additional regions to the packer configuration.
+that deploys to all regions by adding additional regions to the Packertemplate.
```
Comment (jsf9k) [inline README.md]:
```suggestion
that deploys to all regions by adding additional regions to the Packer template.
```

File: variables.pkr.hcl
Comment (jsf9k) [inline variables.pkr.hcl:1]: Should we go ahead and add `nullable = false` for all these variables?

**Review by jsf9k — CHANGES_REQUESTED** [review]

**Review by dav3r — APPROVED** [review]
👍 👍

File: variables.pkr.hcl
Comment (mcdonnnj) [reply variables.pkr.hcl:1]: I don't believe that is a valid [argument for Packer input variables](https://developer.hashicorp.com/packer/docs/templates/hcl_templates/variables#arguments). It's specifically something for Terraform modules as far as I can tell.

**Review by jsf9k — APPROVED** [review]

**Pull request merged**